brave/pom.xml

```diff
@@ -50,4 +64,59 @@
       <scope>test</scope>
     </dependency>
   </dependencies>
+
+  <build>
+    <plugins>
+      <!-- Shade internal types -->
+      <plugin>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>${maven-shade-plugin.version}</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <createDependencyReducedPom>false</createDependencyReducedPom>
+              <artifactSet>
+                <includes>
+                  <include>${brave.groupId}:brave</include>
+                </includes>
+              </artifactSet>
+              <filters>
+                <filter>
+                  <artifact>${brave.groupId}:brave</artifact>
+                  <includes>
+                    <include>brave/internal/ExtraFactory*.class</include>
```

Comment:
these classes will drift on Brave 5.12 which will be added to the next 2.2.x and 3.x version of sleuth.

this means the upgrade order is this first, then again to move to Brave 5.12. Without this PR there's no migration path without a ClassNotFoundException